agent/consul/leader_connect_ca.go

```diff
@@ -1180,8 +1180,7 @@ func (c *CAManager) RenewIntermediate(ctx context.Context, isPrimary bool) error
 		return fmt.Errorf("error parsing active intermediate cert: %v", err)
 	}
 
-	if lessThanHalfTimePassed(c.timeNow(), intermediateCert.NotBefore.Add(ca.CertificateTimeDriftBuffer),
-		intermediateCert.NotAfter) {
+	if lessThanHalfTimePassed(c.timeNow(), intermediateCert.NotBefore, intermediateCert.NotAfter) {
```

Comment:
This was added in #8702 in an attempt to fix this flaky test, but I don't think it accomplished that goal.

I found it much harder to reason about this logic with the extra `.Add()` call, and I think conceptually it's not correct. We really only care about the half time.

I believe the other fixes in this PR will address the flakes.